vscode/src/views/runConfiguration.ts

```diff
@@ -0,0 +1,167 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/* This file has been modified for Oracle Java SE extension */
+
+import * as vscode from 'vscode';
+import { homedir } from 'os';
+import { l10n } from '../localiser';
+
+
+
+
+
+class RunConfigurationNodeProvider implements vscode.TreeDataProvider<vscode.TreeItem> {
+
+	private _onDidChangeTreeData: vscode.EventEmitter<vscode.TreeItem | undefined | null> = new vscode.EventEmitter<vscode.TreeItem | undefined | null>();
+	readonly onDidChangeTreeData: vscode.Event<vscode.TreeItem | undefined | null> = this._onDidChangeTreeData.event;
+
+	refresh(): void {
+		this._onDidChangeTreeData.fire(undefined);
+	}
+
+	getTreeItem(element: vscode.TreeItem): vscode.TreeItem {
+		return element;
+	}
+
+	getChildren(element?: vscode.TreeItem): vscode.ProviderResult<vscode.TreeItem[]> {
+		if (!element) {
+			return [argumentsNode, vmOptionsNode, environmentVariablesNode, workingDirectoryNode];
+		}
+		return [];
+	}
+
+}
+export const runConfigurationNodeProvider = new RunConfigurationNodeProvider();
+
+class RunConfigurationNode extends vscode.TreeItem {
+
+	prompt: string;
+	hint: string;
+	value: string | undefined;
+
+	settingsKey: string;
+
+	constructor(label: string, prompt: string, hint: string, settingsKey: string) {
+		super(label);
+		this.contextValue = 'configureRunSettings';
+		this.collapsibleState = vscode.TreeItemCollapsibleState.None;
+
+		this.prompt = prompt;
+		this.hint = hint;
+
+		this.settingsKey = settingsKey;
+
+		this.value = this.getConfig().get(this.settingsKey);
+		this.updateNode();
+	}
+
+	public configure(_context: vscode.ExtensionContext) {
+		vscode.window.showInputBox(
+			{
+				prompt: this.prompt,
+				value: this.value,
+				placeHolder: this.hint,
+				ignoreFocusOut: true
+			}
+		).then(async val => {
+			if (val !== undefined) {
+				const value = val.toString().trim();
+				this.setValue(value ? value : undefined);
+			}
+		});
+	}
+
+	public getValue(): string | undefined {
+		return this.value;
+	}
+
+	setValue(value: string | undefined) {
+		this.value = value;
+		this.getConfig().update(this.settingsKey, this.value, vscode.workspace.name || vscode.workspace.workspaceFile ? null : true);
+		this.updateNode();
+	}
+
+	updateNode(reload?: boolean) {
+		if (reload) {
+			this.value = this.getConfig().get(this.settingsKey) as string;
+		}
+		this.description = this.value ? this.value : l10n.value("jdk.extension.runConfig.default.label");
+		this.tooltip = `${this.label} ${this.description}`;
+		runConfigurationNodeProvider.refresh();
+	}
+
+	getConfig(): vscode.WorkspaceConfiguration {
+		return vscode.workspace.getConfiguration('jdk.runConfig');
```

Comment:
Is there a constant in *configuration.ts* that can be used in place of the string value `'jdk.runConfig'` ?

Reply:
Done